### ISSUE TITLE
equilibration_v1 leads to error

### ISSUE THREAD

**Issue by piia600** [Acellera/htmd#77]
Hi,

I am trying to equilibrate my system with a standard equilibration procedure from the protocols. However, when running acemd with the output file of the following command, the system produces an error `('Undefined TCL function [calcforces_terminate]')`. 

```
from htmd.protocols.equilibration_v1 import Equilibration
equ = Equilibration()
equ.numsteps = 1000
equ.temperature = 310
equ.write('build/', 'equil/')
```

Unfortunately my skills in TCL are somewhat limited and I can't figure out what is wrong with the input file. Here is also the file generated. 

```
set numsteps 1000
set temperature 310
set refindex {  }
set selindex {  }
set box { 0 0 0 0 0 0 }
set K 0
#
proc flatbot1d {x xm xM K} {
  set f 0
  if {$x < $xm} {
    set f [expr $K*[expr $xm-$x]]
  }
  if {$x > $xM} {
    set f [expr $K*[expr $xM-$x]]
  }
  return $f
}
proc calcforces_init {} {
  global ref sel refindex selindex
  berendsenpressure  off
  set ref [addgroup  $refindex]
  set sel [addgroup  $selindex]
}
proc calcforces {} {
  global ref sel numsteps K box
  loadcoords coords
##FLATBOTTOM
  if {$K>0} {
    set r0 $coords($ref)
    set r1 $coords($sel)
    set dr  [vecsub $r1 $r0]
    set fx [flatbot1d [lindex $dr 0] [lindex $box 0] [lindex $box 1] $K]
    set fy [flatbot1d [lindex $dr 1] [lindex $box 2] [lindex $box 3] $K]
    set fz [flatbot1d [lindex $dr 2] [lindex $box 4] [lindex $box 5] $K]
    #print "dr: $dr  fx: $fx fy: $fy fz: $fz"
    addforce $sel [list $fx $fy $fz]
  }
##EQUIL
  set step [ getstep ]
  if { $step > 500 } {
    berendsenpressure  on
  } else {
    berendsenpressure  off}
  if { $step > [expr $numsteps/2] } {
    constraintscaling 0
  } else {
    constraintscaling [expr 1 + $step*(0.05-1)*2/$numsteps]}
}
proc calcforces_endstep { } { }
#
amber                          	on
berendsenpressure              	on
berendsenpressurerelaxationtime	800
berendsenpressuretarget        	1.01325
celldimension                  	80.13600158691406 94.96400451660156 76.03700256347656
consref                        	structure.pdb
constraints                    	on
constraintscaling              	1.0
coordinates                    	structure.pdb
cutoff                         	9
energyfreq                     	1000
exclude                        	scaled1-4
fullelectfrequency             	2
hydrogenscale                  	4
langevin                       	on
langevindamping                	1
langevintemp                   	$temperature
minimize                       	500
outputname                     	output
parmfile                       	structure.prmtop
pme                            	on
pmegridspacing                 	1.0
restart                        	on
restartfreq                    	5000
rigidbonds                     	all
1-4scaling                     	0.8333333
structure                      	structure.prmtop
switchdist                     	7.5
switching                      	on
tclforces                      	on
temperature                    	$temperature
timestep                       	4
xtcfile                        	output.xtc
xtcfreq                        	25000
run                            	$numsteps
```

Regards,
Piia

**Comment by stefdoerr** [comment]
Input file looks fine. I vaguely remember this being an ACEMD version issue? @mj-harvey what's your input on this?

**Comment by giadefa** [comment]
can you attach the log file?

On 4 July 2016 at 15:17, Stefan <notifications@github.com> wrote:

> Input file looks fine. I vaguely remember this being an ACEMD version
> issue? @mj-harvey <https://github.com/mj-harvey> what's your input on
> this?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Acellera/htmd/issues/77#issuecomment-230288464>, or mute
> the thread
> <https://github.com/notifications/unsubscribe/AHaqOlHR3mcWAPw59yQYt83Gg1-4ul6sks5qSQfngaJpZM4JEVez>
> .
>



-- 
<http://www.acellera.com>

       <https://twitter.com/acellera>
<https://www.youtube.com/user/acelleracom>
<https://www.linkedin.com/company/2133167?trk=tyah&trkInfo=clickedVertical%3Acompany%2CclickedEntityId%3A2133167%2Cidx%3A2-1-2%2CtarId%3A1448018583204%2Ctas%3Aacellera>
    <https://www.acellera.com/md-simulation-blog-news/>
<http://is.gd/1eXkbS>

**Comment by piia600** [comment]
Here is the log.

[equ.txt](https://github.com/Acellera/htmd/files/347284/equ.txt)

**Comment by j3mdamas** [comment]
I would assume this is an acemd issue ratehr than htmd? @mj-harvey?

**Comment by stefdoerr** [comment]
Ok we need to add this function to the new protocols due to a change in ACEMD.

**Comment by j3mdamas** [comment]
I was able to reproduce the error. I fixed it in fd13391072b2c3f3be3e2a437bea132312e0a00f. I tested it with acemd version 2016.5.12 (From conda, but it was not the same as the one from the OP, it was `[] git [222fcc8]`. @mj-harvey?) and version 3212u2, and the fix worked. Can someone (@stefdoerr @mj-harvey) just check if it's OK? If it is, I can close this and do a bugfix version release.

**Comment by j3mdamas** [comment]
Fixed in version 1.2.2, which will be soon on conda. Issue closed.
